drivers/modem/hl7800.c

```diff
@@ -1783,10 +1783,16 @@ static bool on_cmd_atcmdinfo_iccid(struct net_buf **buf, uint16_t len)
 		LOG_INF("EID: %s", delim + 1);
 	}
 
+/* Silence format-truncation warnings when logs are not enabled, as the
+ * compiler seems to think that snprintk truncation is not handled.
+ */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
 	ret = snprintk(iface_ctx.mdm_iccid, sizeof(iface_ctx.mdm_iccid), "%s", value);
```

Comment:
Could this just use strncpy instead?

Reply:
I'll be honest I partially did not want to change the function code for the sake of skipping the warning and partially was afraid of messing up the explicit string null termination :-) but I realize now that there's load of other strncpy in this file anyway so I guess I'll give it a go.